CriticalMass/UpdateDataOperation.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  UpdateDataOperation.swift
+//  CriticalMaps
+//
+//  Created by Илья Глущук on 13/10/2019.
+//  Copyright © 2019 Pokus Labs. All rights reserved.
+//
+
+import Foundation
+
+final class UpdateDataOperation: AsyncOperation {
+    private struct SendLocationPostBody: Encodable {
+        let device: String
+        let location: Location
+    }
+
+    private let locationProvider: LocationProvider
+    private let idProvider: IDProvider
+    private let networkLayer: NetworkLayer
+
+    var result: (Result<ApiResponse, NetworkError>)?
+
+    init(locationProvider: LocationProvider, idProvider: IDProvider, networkLayer: NetworkLayer) {
+        self.locationProvider = locationProvider
+        self.idProvider = idProvider
+        self.networkLayer = networkLayer
+        super.init()
+    }
+
+    override func main() {
+        guard let currentLocation = locationProvider.currentLocation else {
+            let request = GetLocationsAndChatMessagesRequest()
+            networkLayer.get(request: request) { [weak self] result in
+                guard let self = self else { return }
+
+                self.result = result
+                self.completeOperation()
+            }
+
+            return
+        }
```

Comment:
having a lot of logic in the else case and having two nested guards makes it hard to read for me. Maybe we can find a cleaner solution.

Reply:
Agree. Moved code to separate method